src/commands/update.rs

```diff
@@ -0,0 +1,90 @@
+use clap::ArgMatches;
+use log::{debug, error, info};
+
+use std::process::exit;
+
+use crate::server;
+
+const EXIT_NO_UPDATE_AVAILABLE: i32 = 0;
+const EXIT_UPDATE_AVAILABLE: i32 = 1;
+
+enum Action {
+  Check,
+  Force,
+  Regular,
+}
+
+impl Action {
+  fn new(check: bool, force: bool) -> Self {
+    match (check, force) {
+      (true, true) => panic!("`check` and `force` are mutually exlusive!"),
+      (true, false) => Self::Check,
+      (false, true) => Self::Force,
+      (false, false) => Self::Regular,
+    }
+  }
+}
+
+pub fn invoke(args: &ArgMatches) {
+  info!("Checking for updates");
+
+  if !server::is_installed() {
+    error!(
+      "Failed to find server executable. Can't update! If the server isn't installed yet then you \
+        likely need to run `odin install`."
+    );
+    exit(1);
+  }
+
+  let dry_run = args.is_present("dry_run");
+  let check = args.is_present("check");
+  let force = args.is_present("force");
+
+  let server_running = server::is_running();
+  let update_available = server::update_is_available();
+  if update_available {
+    info!("A server update is available!");
+  } else {
+    info!("No server updates found");
+  }
+
+  match Action::new(check, force) {
```

Comment:
This section is a little hard to read, do you think we could typecast `dry_run, update_available` into a set of options as its own construct?

Reply:
Can you expand a bit more on what you mean? I'm not quite sure I'm understanding you here

Reply:
Do you mean like this?

```rust
struct UpdateContext {
    dry_run: bool,
    update_available: bool,
    server_running: bool,
}
```

Reply:
Ended up going with separate enums for each one since the struct was messy at best. I think it worked out pretty well!

Reply:
Changes implemented in 331506beed5b88706a32e82a0f7878d1c40104c1! I ended up going with an enum for each of `dry_run`, `update_available`, and `server_running` as well as breaking out the usage usage of checking for an update, forcing an update, and performing a regular update into separate functions to group things a bit better. I think it's a lot more readable now! :D

Lmk what you think!